requirements-optional.txt

```
@@ -3,3 +3,5 @@ toml<0.11
 pytest-cov
 coverage<8.0.0
 ipykernel
+# Remember to update the CI pyright versions when updating this line
+pyright<1.1.300
```

```suggestion
pyright<1.2
```

It is too much to check every minor version. This will ping every week, and there is no real reason to do that.

---

https://github.com/microsoft/pyright/releases

---

Actually, Pyright breaks on their minor versions, and this caused us problems before.
So here it's better to have `<1.1.300` I think.

---

Maybe <1.1.3** ?